cardano-db-sync/src/Cardano/DbSync/Era/Shelley/Generic/Tx/Alonzo.hs

```diff
@@ -289,8 +291,10 @@ mkTxScript (hsh, script) =
     plutusCborScript =
       case script of
         Alonzo.TimelockScript {} -> Nothing
-        Alonzo.PlutusScript _lang s ->
+        Alonzo.PlutusScript Alonzo.PlutusV1 s ->
           Just . Api.serialiseToCBOR . Api.PlutusScript Api.PlutusScriptV1 $ Api.PlutusScriptSerialised s
+        Alonzo.PlutusScript Alonzo.PlutusV2 s ->
+          Just . Api.serialiseToCBOR . Api.PlutusScript Api.PlutusScriptV2 $ Api.PlutusScriptSerialised s
```

Comment:
Just a personal note, this is a nice catch and intially I thought it fixes a big bug. But it doesn't change the behaviour, because the serialiser actually ignores the version tag.
```
instance IsScriptLanguage lang => SerialiseAsCBOR (Script lang) where
    serialiseToCBOR (SimpleScript SimpleScriptV1 s) =
      CBOR.serialize' (toShelleyMultiSig s)

    serialiseToCBOR (SimpleScript SimpleScriptV2 s) =
      CBOR.serialize' (toAllegraTimelock s :: Timelock.Timelock StandardCrypto)

    serialiseToCBOR (PlutusScript PlutusScriptV1 s) =
      CBOR.serialize' s

    serialiseToCBOR (PlutusScript PlutusScriptV2 s) =
      CBOR.serialize' s
```

Reply:
Good find! I think it's worth a comment.